mrp_workorder_sequence/hooks.py

```diff
@@ -0,0 +1,11 @@
+# Copyright 2022 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl)
+from odoo import SUPERUSER_ID, api, tools
+
+
+def post_init_hook(cr, registry):
+    env = api.Environment(cr, SUPERUSER_ID, {})
+    all_workorders = env["mrp.workorder"].search([], order="production_id ASC, id ASC")
+    for _, workorders in tools.groupby(all_workorders, lambda w: w.production_id):
+        for seq, wo in enumerate(workorders, 1):
+            wo.sequence = seq
```

Comment:
I'm worried this might cause a `MemoryError` on large databases.
It's common with this kind of scripts.

The best solution to this is to use raw sql whenever possible, as it will perfom much faster.
Maybe something like this (⚠️ may need fixing, untested!)

```suggestion
    cr.execute(
        """
            UPDATE mrp_workorder w
            SET w.sequence = n.sequence
            FROM (
                SELECT
                    id,
                    ROW_NUMBER() OVER (PARTITION BY production_id) AS sequence
                FROM mrp_workorder
                ORDER BY production_id, id
            ) AS n
            WHERE w.id = n.id
        """
    )
```

----

Alternatively, look into openupgradelib's `chunked` method

Reply:
@ivantodorovich That looks quite good actually, I agree it's less risky to raise a MemoryError